ArduPlane/GCS_Mavlink.cpp

```diff
@@ -794,13 +794,169 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_do_reposition(const mavlink_com
     return MAV_RESULT_FAILED;
 }
 
+// these are GUIDED mode commands that are RATE or slew enabled, so you can have more powerful control than default controls.
+MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_guided_slew_commands(const mavlink_command_int_t &packet)
+{
+  switch(packet.command) {
+    
+#if OFFBOARD_GUIDED == ENABLED
+    case MAV_CMD_GUIDED_CHANGE_SPEED: {
+        // command is only valid in guided mode
+        if (plane.control_mode != &plane.mode_guided) {
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+         // only airspeed commands are supported right now...
+        if (int(packet.param1) != SPEED_TYPE_AIRSPEED) {  // since SPEED_TYPE is int in range 0-1 and packet.param1 is a *float* this works.
+            return MAV_RESULT_UNSUPPORTED;
+        }
+
+         // reject airspeeds that are outside of the tuning envelope
+        if (packet.param2 > plane.aparm.airspeed_max || packet.param2 < plane.aparm.airspeed_min) {
+            return MAV_RESULT_UNSUPPORTED;
+        }
+
+         // reject duplicate airspeeds
+        float new_target_airspeed_cm = packet.param2 * 100;
+//        if (new_target_airspeed_cm == plane.guided_state.target_airspeed_cm) {
+        if ( fabsf(new_target_airspeed_cm - plane.guided_state.target_airspeed_cm)<0.001f) { // compare if two floats are 'the same'
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+        plane.guided_state.target_airspeed_cm = new_target_airspeed_cm;
+        plane.guided_state.target_airspeed_time_ms = AP_HAL::millis();
+
+         if (is_zero(packet.param3)) {
+            // the user wanted /maximum acceleration, pick a large value as close enough
+            plane.guided_state.target_airspeed_accel = 1000.0f;
+        } else {
+            plane.guided_state.target_airspeed_accel = fabsf(packet.param3);
+        }
+
+         // assign an acceleration direction
+        if (plane.guided_state.target_airspeed_cm < plane.target_airspeed_cm) {
+            plane.guided_state.target_airspeed_accel *= -1.0f;
+        }
+        return MAV_RESULT_ACCEPTED;
+    }
+
+     case MAV_CMD_GUIDED_CHANGE_ALTITUDE: {
+        // command is only valid in guided
+        if (plane.control_mode != &plane.mode_guided) {
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+        // command is only valid if alt is not at it's default value of -1. this is a float, so the use of is_zero(x-1)  
```

Comment:
the comment implies that -1 is the only invalid value, but implementation means that only invalid value is 1. Which is it?
Also, please use is_equal() instead of munging with is_zero()
I think what you intended is:
  if (is_equal(packet.z, -1))

Reply:
will fix, to both... although '-1' is an int not a float so won't work exactly as u typed, needs -1.0f